### PR TITLE
Rebalancing mining sand

### PR DESCRIPTION
**Pull request by Sindorman**
This PR makes these:

- You can no longer dig more than one sand turf at the time with shovel/drill.

- Digging sand times are reduces for all tools. Pickaxe - 0.67 seconds(It takes same amount of time to mine it now as before with queuing 3x3), drill/borg - 0.3 second, Diamond drill - instant
- Reduced fine for accidentally burning ore into sluggs from 3 times to only +25% of what ore was worth.

Feedback thread: https://forums.aurorastation.org/topic/12466-sand-digging-rebalance/

### CODE REVIEW

**Comment by Sindorman** [comment]
!review

**Comment by FreshRefreshments** [comment]
This isn't rebalancing, this is a huge nerf. Being able to dig in a 3 by 3 area speeds up the process of digging sand to the same amount of time it takes to dig out of rocks, and i've got experience to show for this. It is now 9x slower because you need to do them one by one, instead of being able to clear out a square at a time. It doesnt matter how much you reduce that individual dig, because you can no longer spread it out among a larger area. 
Slagging is a huge downside because its meant to deter you. You aren't supposed to slag stuff, and its easy to avoid it. The settings are the same every time. 
Mining doesn't need these huge nerfs and "rebalances" that aren't necessary. Things like greatly increasing the amount of time people dig in sand greatly increases the amount of time science goes without materials, which is already a problem.

**Comment by Conspiir** [comment]
Changing shovels is quite unnecessary. 

I know, I know, this isn't a feedback thread. But this is a pretty intense nerfing of two departments. Shovels do the job well. The thing is, it takes a long time to do it. A miner who chooses to shovel and a miner who chooses to go out and pickaxe dig come in in the same amount of time and with the same average number of points, provided equal skill level. There is a difference in _what_ they come in with (pick diggers usually have a lot more of one or two or three ores, while shovel diggers have very very small amounts of everything, including hydrogen wafers which are pretty useless). Shovel digging can get research a tiny amount of every material so they can print a few things before running out and having to wait for the next load. This change will make research and mining take an overall longer amount of time to get up and going. It already takes at least 30 minutes, I don't think either side wants it to take longer.

With the new KA mining is set to get, pick mining is already becoming better than shovel mining and I foresee KA miners (formerly pick miners) having a far better time than shovel miners even if shovels are left as-is. But if you make digging with a shovel a lot _faster_, it will become better than a pick or Class E. If you make it only able to dig one turf at a time, it becomes useless, straight-up. The balance is in how slow it is to get a wider variety (but smaller number) of materials for your scientific slavedrivers, and to get enough real estate to do so. As far as I see, there's nothing "overly strong" or "gamebreaking" about mining. It doesn't seem to hurt anyone else's fun, or valid antags, or detract from the game environment.

The slag change is good, because no experienced miner makes those types of mistakes anyway and it is only a boon to new players that might make a mistake.

**Comment by Sindorman** [comment]
This is not a feedback thread. So yes, please stick to the code issues. If you want a feedback thread please request one.

To answer your questions:
It makes no sense how a single handed shovel can dig 3x3 tiles area. Secondly shovel was never meant to be at the same level as using KA to mine rock, using sad with shovel is supposed to be early game/last restort, we have drills for mining lots of stuff fast. Hand drills are getting buff to digging and you got a big industrial drill that mines on its own, which is faster.

This is a ground work to adding automation to mining where you can setup drill and pipe network and watch that ore flow to the station.

**Comment by Conspiir** [comment]
I'm. I just. Really going to have to request a feedback thread. There is so much wrong with what you've proposed, I'm doubting you've played much mining.

Using a shovel IS early game/last resort. I can't imagine getting to the point where I can afford a diamond drill and still be mining _sand_ for any sort of value. Gives me the vapors just thinking about it.

**Comment by Sindorman** [comment]
Thread posted

File: code/modules/mining/mine_turfs.dm [hunk 1]
```diff
@@ -648,30 +648,36 @@ var/list/asteroid_floor_smooth = list(
...
-	if(is_type_in_typecache(W, usable_tools))
+	var/static/list/usable_tools = list(
+		/obj/item/weapon/shovel = 25,
+		/obj/item/weapon/pickaxe/diamonddrill = 5,
+		/obj/item/weapon/pickaxe/drill = 10,
+		/obj/item/weapon/pickaxe/borgdrill = 10
+	)
+
+	var/time = usable_tools[W.type]
+	message_admins(time)
```
Comment (skull132) [inline code/modules/mining/mine_turfs.dm]:
```suggestion
```

File: code/modules/mining/mine_turfs.dm [hunk 2]
```diff
@@ -648,30 +648,35 @@ var/list/asteroid_floor_smooth = list(
...
+	)
+
+	var/time = usable_tools[W.type]
+	if(time)
 		var/turf/T = user.loc
 		if (!(istype(T)))
 			return
 		if(digging)
 			return
+		if(W:digging)
```
Comment (MarinaGryphon) [inline code/modules/mining/mine_turfs.dm]: Noooooooooo. No? No. Absolutely not. We don't do this here.

File: code/modules/mining/mine_turfs.dm [hunk 3]
```diff
@@ -648,30 +648,35 @@ var/list/asteroid_floor_smooth = list(
...
 				return
 			to_chat(user, "<span class='warning'>You start digging deeper.</span>")
 			playsound(user.loc, 'sound/effects/stonedoor_openclose.ogg', 50, 1)
 			digging = 1
-			if(!do_after(user, 60))
+			W:digging = TRUE
+			if(!do_after(user, time + 10))
 				if (istype(src, /turf/unsimulated/floor/asteroid))
 					digging = 0
+				W:digging = FALSE
```
Comment (MarinaGryphon) [inline code/modules/mining/mine_turfs.dm]: Noooooooo. We do typechecks and casting here.

File: code/modules/mining/mine_turfs.dm [hunk 4]
```diff
@@ -648,30 +648,35 @@ var/list/asteroid_floor_smooth = list(
...
 				return
-
+			W:digging = FALSE
```
Comment (MarinaGryphon) [inline code/modules/mining/mine_turfs.dm]: Nooooooooooooooo! No no no no no no! No. Typecheck and cast.

File: code/modules/mining/mine_turfs.dm [hunk 5]
```diff
@@ -716,10 +721,13 @@ var/list/asteroid_floor_smooth = list(
 		playsound(user.loc, 'sound/effects/stonedoor_openclose.ogg', 50, 1)
 
 		digging = 1
-		if(!do_after(user,40))
+		W:digging = TRUE
```
Comment (MarinaGryphon) [inline code/modules/mining/mine_turfs.dm]: See above.

File: code/modules/mining/mine_turfs.dm [hunk 6]
```diff
@@ -716,10 +721,13 @@ var/list/asteroid_floor_smooth = list(
...
+		if(!do_after(user, time))
 			if (istype(src, /turf/unsimulated/floor/asteroid))
 				digging = 0
+			W:digging = FALSE
```
Comment (MarinaGryphon) [inline code/modules/mining/mine_turfs.dm]: See above.

File: code/modules/mining/mine_turfs.dm [hunk 7]
```diff
@@ -716,10 +721,13 @@ var/list/asteroid_floor_smooth = list(
...
 			return
+		W:digging = FALSE
```
Comment (MarinaGryphon) [inline code/modules/mining/mine_turfs.dm]: See above.

**Pull request closed without merging**